### PR TITLE
APP-5909

### PR DESCRIPTION
**Pull request by nicksanford**
[Ticket](https://viam.atlassian.net/browse/APP-5909)

Change viam cli's `viam data database configure` command to require user confirmation before executing change to database auth credentials if a user already exists.

```bash
# first time configuring
$ viam data database configure --org-id <REDACTED> --password <REDACTED>

# second time configuring, with the user aborting
Configured database user for org <REDACTED>
$ viam data database configure --org-id <REDACTED> --password <REDACTED>
WARNING!!

You or someone else in your organization have already created a user.

The following steps update the password for that user.

Once you have updated the password, you will need to update all dashboards or
other integrations relying on this password.

Do you want to continue?

Continue: y/n
n
Error: Aborted
exit status 1

# third time attempting to 
$ viam data database configure --org-id <REDACTED> --password <REDACTED>
WARNING!!

You or someone else in your organization have already created a user.

The following steps update the password for that user.

Once you have updated the password, you will need to update all dashboards or
other integrations relying on this password.

Do you want to continue?

Continue: y/n
y
Configured database user for org <REDACTED>
```

### CODE REVIEW

File: cli/data.go [hunk 1]
```diff
@@ -903,6 +903,31 @@ func (c *viamClient) dataRemoveFromDataset(datasetID, orgID, locationID string,
 	return nil
 }
 
+// DataConfigureDatabaseUserConfirmation is the Before action for 'data database configure'.
+// it asks for the user to confirm that they are aware that they are changing the authentication
+// credentials of their database
+func DataConfigureDatabaseUserConfirmation(c *cli.Context) error {
+	yellow := "\033[1;33m%s\033[0m"
+	printf(c.App.Writer, yellow, "WARNING!!!")
+	printf(c.App.Writer, yellow, "this will change the password of the database")
```
Comment (npentrel) [inline cli/data.go:912]:
```suggestion
	printf(c.App.Writer, yellow, "If you or someone else in your organization have already created this user, the following steps update the password for that user instead. Dashboards or other integrations relying on this password will then need to be updated.")
```

File: cli/data.go [hunk 2]
```diff
@@ -903,6 +903,31 @@ func (c *viamClient) dataRemoveFromDataset(datasetID, orgID, locationID string,
...
+	printf(c.App.Writer, yellow, "do you want to continue?")
```
Comment (npentrel) [inline cli/data.go:913]:
```suggestion
	printf(c.App.Writer, yellow, "Do you want to continue?")
```

File: cli/data.go [hunk 3]
```diff
@@ -903,6 +903,31 @@ func (c *viamClient) dataRemoveFromDataset(datasetID, orgID, locationID string,
 	return nil
 }
 
+// DataConfigureDatabaseUserConfirmation is the Before action for 'data database configure'.
+// it asks for the user to confirm that they are aware that they are changing the authentication
+// credentials of their database
+func DataConfigureDatabaseUserConfirmation(c *cli.Context) error {
+	yellow := "\033[1;33m%s\033[0m"
+	printf(c.App.Writer, yellow, "WARNING!!!")
+	printf(c.App.Writer, yellow, "If you or someone else in your organization have already created this user, the following steps update the password for that user instead. Dashboards or other integrations relying on this password will then need to be updated.")
```
Comment (npentrel) [inline cli/data.go:912]: If we can check if a database usere has already been configured, then I'd only print this in that case:
```suggestion
	printf(c.App.Writer, yellow, "You or someone else in your organization have already created a user. The following steps update the password for that user. Once you have updated the password, you will need to update all dashboards or other integrations relying on this password.")
```

**Review by npentrel — APPROVED** [review]
Looks great. Will the cli automatically wrap the text or should we maybe add new line characters?

**Comment by nicksanford** [comment]
> Looks great. Will the cli automatically wrap the text or should we maybe add new line characters?

I've improved the whitespace handling.

**Pull request merged**